cockatrice/src/tab_debuglog.cpp

```diff
@@ -1,35 +1,48 @@
-#include "dlg_viewlog.h"
+#include "tab_debuglog.h"
 #include "logger.h"
 
 #include <QVBoxLayout>
 #include <QPlainTextEdit>
 
-DlgViewLog::DlgViewLog(QWidget *parent)
-: QDialog(parent)
+TabDebugLog::TabDebugLog(TabSupervisor *tabSupervisor, QWidget *parent)
```

Comment:
I agree they are too tightly coupled and it would be better to break this link.
Currently `Tab` requires a `TabSupervisor` in order to use it to get an handle to informations related to the server connection, eg:
 * `tabSupervisor->getIsLocalGame()`
 * `tabSupervisor->getUserListsTab()->getIgnoreList()->getUsers().contains(senderName)`
 * `TabRoom *r = tabSupervisor->getRoomTabs().value(game.room_id());`

Imho `TabSupervisor` should only handle the collection of tabs and, and not be the central dispatcher informations about the current server connection.
Reworking this in a saner way could also lead into removing some limitations, eg. not being able to open a local game when connected to a server.